pandas/tests/indexing/test_loc.py

```diff
@@ -2618,6 +2618,16 @@ def test_loc_indexer_all_false_broadcast(self):
         df.loc[np.array([False], dtype=np.bool_), ["a"]] = df["b"]
         tm.assert_frame_equal(df, expected)
 
+    def test_loc_mask_dtype_change(self):
+        # GH: 29707
+        df = DataFrame([])
+        df["col0"] = np.array([1, 4])
+        df["col1"] = np.array([6, 5], dtype=object)
+        expected = df.copy()
+        with pd.option_context("mode.chained_assignment", None):
+            df.col1.loc[df.col0 > 10] = "string"
+        tm.assert_frame_equal(df, expected)
```

Comment:
this test is throwing an error
```
FAILED pandas/tests/indexing/test_loc.py::TestLocBooleanMask::test_loc_mask_dtype_change - pandas.errors.ChainedAssignmentError: A value is trying to be set on a copy of a DataFrame or Series through chained assignment.
When using the Copy-on-Write mode, such chained assignment never works to update the original DataFrame or Series, because the intermediate object on which we are setting values always behaves as a copy.

Try using '.loc[row_indexer, col_indexer] = value' instead, to perform the assignment in a single step.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

```